### PR TITLE
Generate a pom.xml

### LINKED ISSUE #131: Generate pom.xml
For now, just make it all runnable from the CLI.

Still TBD: how/whether we sync back and forth from the pom.xml to Eclipse and whether the Eclipse Build goes through maven

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis @patflynn 

fix #131

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/templates/pom.xml.ftl [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                        <directory>${basedir}/src/main/webapp/WEB-INF</directory>
+                        <filtering>true</filtering>
+                        <targetPath>WEB-INF</targetPath>
+                    </resource>
+                </webResources>
+            </configuration>
+        </plugin>
+
+        <plugin>
+            <groupId>com.google.appengine</groupId>
```
Comment (patflynn) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/templates/pom.xml.ftl]: you'll eventually want to update this to the new maven plugins that we're going to release to dogfood soon.

**Comment by briandealwis** [comment]
I only gave a very cursory look, but this looks like creating a Maven archetype project.  Rather than reinvent that wheel,  I think we should have different streams for creating a new project:

- Create a native new WTP AppEngine project, with appropriate facets.  This is pretty much the path you're following right now, though it should be transitioned to using faceted projects, and I realize that some of the code generation you're doing should be transitioned into facet actions instead.
- Create a new Maven AppEngine project
   - This should hook into the M2Eclipse support and creates a project using the appropriate archetype
   - Using the new Maven Project wizard is painful as it seemingly doesn't come configured out of the box to use the Maven Central Archetype Catalog, which is baffling
   - and there are several appengine archetypes and it's not clear which is the right one, so there's definite value here
- Create a new Gradle AppEngine project

**Comment by elharo** [comment]
This PR isn't the final build system. The team asked me to throw in a quick pom.xml so we could have a CLI build in place for now. #134 and #118 are about getting more of this working inside Eclipse. I think we're leaning against making the build inside Eclipse maven based, though that could change with a strong enough case in favor or maven. If we do go all the way down the maven road, we'll need to understand how changes in the pom.xml sync back into the Eclipse build.

One question about facets: do these indeed work in regular Eclipse for Java? I.e. not Eclipse for Java EE? From what I'm reading they seem like an EE thing. 

Gradle we're still discussing in #132. How much do you think Eclipse developers use Gradle? Our impression is that it's more common in the IntelliJ space, but it's easy enough to add a build.gradle file in addition to the pom and everything else.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/templates/pom.xml.ftl (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/templates/pom.xml.ftl]: For sure, easy enough to do when they're ready.

**Comment by briandealwis** [comment]
LGTM

**Comment by briandealwis** [comment]
> One question about facets: do these indeed work in regular Eclipse for Java? I.e. not Eclipse for Java EE? From what I'm reading they seem like an EE thing.

Faceted Projects are a WTP feature; they never migrated further into the platform.  They're almost equivalent to natures, but natures don't have the ability to place constraints like pre-reqs and anti-reqs.  There's nothing preventing using a faceted project for a Java project, but I guess it's rare for non-WTP projects to mix different languages and libraries.

**Pull request merged**